nntrainer/layers/pooling2d_layer.cpp

```diff
@@ -181,21 +187,20 @@ void Pooling2DLayer::calcDerivative(RunLayerContext &context) {
     }
   } break;
   case PoolingType::global_max: {
-    const int *iter = pool_helper.getData<int>();
     float *deriv_data = deriv.getData();
-    unsigned int outer_loop = batch * channel;
-    for (unsigned int _ = 0; _ < outer_loop; _++) {
-      for (unsigned int i = 0; i < out_map_size; ++i) {
-        float der = *deriv_data / pool_helper.length();
-
-        for (unsigned int idx = 0; idx < pool_helper.length(); idx++) {
+    for (unsigned int b = 0; b < batch; b++) {
+      for (unsigned int c = 0; c < channel; c++) {
+        const int *iter =
+          pool_helper.getData<int>() + pool_helper.getIndex(b, c, 0, 0);
```

Comment:
Note: This will be working for non-contiguous tensor but there is additional overhead of calculating the output index. It was the reason I tried to not put batch argument to this function.

I guess it is fine until we do some benchmarks.

Reply:
When using std::vector, the batch and channel dimensions were merged. The same can be done with tensor, where the batch dimension would become batch * channel size. is that what you meant?

Reply:
Yeah, sth like that.

Reply:
I will do this in a separte small PR.